boilerplates/react-union-boilerplate-ssr-basic/src/apps/SampleApp/components/Root/Root.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Union } from 'react-union';
+import { hot } from 'react-hot-loader';
+
+import routes from '../../routes';
+
+import './Root.css';
+
+const Root = ({ isServer }) => <Union routes={isServer ? null : routes} />;
```

Comment:
In SSR context, the scanning occurs outside `<Union />`, so passing routes here would have no effect. I added an invariant to the Union component itself to prevent users from passing routes here, but it adds unnecessary logic to the application (checking if the Root component is being rendered in SSR context).

It would make more sense to pass routes in `src/index.js`, as that file is client-side only, but then we'd lose HMR for routes 😞 Would it make more sense to always pass the routes to `<Union />` even though they would be ignored?

Or we could get absolutely crazy and not pass routes to the SSR server directly, but attempt to get them by traversing the React tree and finding a Union component 😄

Reply:
in matter of fact, <Union /> component on server is not needed at all... ?

Reply:
You mean the Union component? I don't think that's viable. It still handles rendering of our widgets and it wouldn't make sense if we wanted to wrap the application e.g. with a store provider (you'd have to use some kind of placeholder element to substitute for `<Union />`).

Reply:
I would pass them to the `<Union />` component and just ignore them. It is more friendly.

Reply:
Alright, will do :)